### ISSUE TITLE
avoid page refresh - just load new content & source

### ISSUE THREAD

**Issue by wilzbach** [stonemaster/dlang-tour#152]
actually I thought it is a SPA until I had a look at the debugger

**Comment by stonemaster** [comment]
Due to caching a next page load is just 4KB. Is that really needed? What are the benefits. I currently don't see any latency issues. That would mean a complete overhaul of the current website structure and more logic in javascript that is moved from the vibe.d backend.

**Comment by wilzbach** [comment]
> Due to caching a next page load is just 4KB. Is that really needed? What are the benefits. I currently don't see any latency issues. 

Yes, we have an average page load of 4.8 seconds for every page even though we could get easily to 0 with a bit of JS.

The GoTour downloads the entire "lesson" with all pages as json in the beginning - it's just 27kB (gzipped).

https://tour.golang.org/lesson/

That might an idea worth trying? Otherwise we could split the requests into the chapters.

>  and more logic in javascript that is moved from the vibe.d backend.

Yes I don't like this either, but I think 5 seconds are too long for a user to wait.
